### DISCUSSION TITLE
How can I take a map as input parameter using dynamic-graphql

### DISCUSSION THREAD

**Discussion by shivamka1** [smmoosavi/dynamic-graphql#23] [Q&A]
I want to pass a json ocject as input parameter to a gql api. I can't any test examples that deal with a more complex input parameter than [a simple flat data structure](https://github.com/smmoosavi/dynamic-graphql/blob/main/derive/tests/input_object/type_tests.rs). 

The json parameter could be `HashMap<String, HashMap<String, Value>>`. For instance, `HashMap<EmpId, HashMap<Key, Value>>` i.e., properties of employees. Or something like:
```rust
#[derive(InputObject, Debug)]
pub(crate) struct EmpProperty {
    pub key: String,
    pub value: Value,
}

#[derive(InputObject, Debug)]
pub(crate) struct Employees {
    pub coordinates: HashMap<String, Vec<EmpProperty>>,
}
```

Please suggest.

**Comment by smmoosavi** [comment]
Hi,

I added examples for `JsonValue`, `JsonObject`, and `KeyValue` scalar in #24. Hope it helps!
